ecosystem/typescript/sdk/packages/aptosClient/src/index.node.ts

```diff
@@ -0,0 +1,106 @@
+import got, { OptionsOfJSONResponseBody, RequestError, Response } from "got";
+import { CookieJar } from "./cookieJar";
+import { AptosClientRequest, AptosClientResponse } from "./types";
+
+const cookieJar = new CookieJar();
+
+export default async function aptosClient<Res>(requestOptions: AptosClientRequest): Promise<AptosClientResponse<Res>> {
+  const { params, method, url, headers, body } = requestOptions;
+
+  const request: OptionsOfJSONResponseBody = {
+    http2: true,
+    searchParams: convertBigIntToString(params),
+    method,
+    url,
+    responseType: "json",
+    headers,
+    hooks: {
+      beforeRequest: [
+        (options) => {
+          const cookies = cookieJar.getCookies(new URL(options.url!));
+
+          if (cookies?.length > 0 && options.headers) {
+            /* eslint-disable no-param-reassign */
+            options.headers.cookie = cookies.map((cookie: any) => `${cookie.name}=${cookie.value}`).join("; ");
```

Comment:
Do we grab the cookie associated with an request url, and then assign it to the current request? Is this to ensure the cookie associated with the url is always included in the headers?

Reply:
yes. this is just our own implementation of a cookieJar. it is an existing code in our current sdk, where axios doesnt support cookies in a node env, we have our own cookie implementation. this is the same issue here, where got doesn't support cookies out of the box so we use our implementation. The need for cookie in the request is b/c sdk depends on the cookie based sticky routing that happens in the LB